### ISSUE TITLE
Fails to parse INSERT INTO ... RETURNING statements

### ISSUE THREAD

**Issue by loafofpiecrust** [sqlfluff/sqlfluff#2693]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I tried to lint a query that inserts into a table and returns the id column value of the inserted row. [This is postgres syntax for doing that](https://www.postgresql.org/docs/9.5/dml-returning.html). It failed with an error. I also noticed that writing a CTE that uses `INSERT INTO ... RETURNING` fails to parse because I guess sqlfluff assumes CTEs are `SELECT`?

### Expected Behaviour

I expect sqlfluff to support common (I think) postgres patterns like `INSERT INTO ... RETURNING`.

### Observed Behaviour

I've got a few other linting errors with my file, but the last one is the important one.

```sh
❯ sqlfluff lint types/queries/insert_word_in_document.sql 
== [types/queries/insert_word_in_document.sql] FAIL                                                   
L:   1 | P:  99 | L016 | Line is too long
L:   3 | P:   3 | L003 | Indentation not consistent with line #1
L:   3 | P:  56 |  PRS | Line 3, Position 99: Found unparsable section: '
                       | RETURNING id'
All Finished 📜 🎉!
```

### How to reproduce

Here's the `.sql` file I'm trying to lint:

```sql
INSERT INTO word (source_text, simple_phonetics)
  VALUES ($1, $2)
RETURNING id
```

### Dialect

postgres

### Version

```
sqlfluff, version 0.10.0
```
```
Python 3.9.10
```

### Configuration

```ini
[sqlfluff]
dialect = postgres
templater = placeholder
exclude_rules = L031, L057

[sqlfluff:indentation]
indented_joins = True

[sqlfluff:rules]
tab_space_size = 2

[sqlfluff:templater:placeholder]
param_style = numeric_dollar
1 = 'one'
2 = 'two'
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
